pkg/3scale/amp/component/highavailability_options.go

```diff
@@ -0,0 +1,29 @@
+package component
+
+import "github.com/go-playground/validator/v10"
+
+type HighAvailabilityOptions struct {
+	AppLabel                            string `validate:"required"`
+	BackendRedisQueuesEndpoint          string `validate:"required"`
+	BackendRedisQueuesSentinelHosts     string `validate:"required"`
```

Comment:
The "arbitrary settings" are the symptoms of a bad smell, not the cause. 

IMO The fields of the options struct should be checked against not being set or whatever validation are needed. Removing validation because for templates they are not needed does not seems to me a good way to go. 

We should refactor the templates usage of HighAvailability option and you can have the best of both worlds: validation and not having to set "arbitrary" values. wdyt?